src/test/codewhisperer/service/telemetry.test.ts

```diff
@@ -0,0 +1,655 @@
+        isShowMethodsEnabled: true,
+        isManualTriggerEnabled: true,
+        isAutomatedTriggerEnabled: false,
+        isSuggestionsWithCodeReferencesEnabled: true,
+    }
+
+    beforeEach(async function () {
+        sandbox = sinon.createSandbox()
+        tempFolder = (await createTestWorkspaceFolder()).uri.fsPath
+        client = mockClient()
+        sandbox.stub(AuthUtil.instance, 'isConnected').returns(true)
+        sandbox.stub(AuthUtil.instance, 'isConnectionExpired').returns(false)
+        await resetStates()
+    })
+
+    afterEach(async function () {
+        sandbox.restore()
+        await resetStates()
+    })
+
+    async function resetStates() {
+        await RecommendationHandler.instance.clearInlineCompletionStates()
+        resetCodeWhispererGlobalVariables()
+    }
+
+    function mockClient(): DefaultCodeWhispererClient {
+        const response11 = aResponse(
+            'session_id_1',
+            'request_id_1',
+            'fake-nextToken',
+            { content: 'Foo' },
+            { content: 'Bar\n\nBar' }
+        )
+        const response12 = aResponse('session_id_1', 'request_id_1_2', undefined, { content: 'FooFoo' })
+        const response2 = aResponse(
+            'session_id_2',
+            'request_id_2',
+            undefined,
+            { content: 'Baz\nBaz' },
+            { content: 'Baz' }
+        )
+        const response3 = aResponse('session_id_3', 'request_id_3', undefined, { content: 'Qoo' })
+
+        const cwClient = new DefaultCodeWhispererClient()
+        const stub = sandbox.stub(cwClient, 'listRecommendations')
+        stub.onCall(0).resolves(response11)
+        stub.onCall(1).resolves(response12)
+        stub.onCall(2).resolves(response2)
+        stub.onCall(3).resolves(response3)
+
+        return cwClient
+    }
+
+    function assertSessionClean() {
+        assert.strictEqual(session.requestIdList.length, 0)
+        assert.strictEqual(session.recommendations.length, 0)
+        assert.strictEqual(session.completionTypes.size, 0)
+        assert.strictEqual(session.completionTypes.size, 0)
+    }
+
+    it('this test should be removed', async function () {
+        const r = (await client.listRecommendations(aRequest())) as any
+        const r2 = (await client.listRecommendations(aRequest())) as any
+
+        assert.strictEqual(r.recommendations.length, 2)
+        assert.strictEqual(r2.recommendations.length, 1)
+
+        assert.ok(AuthUtil.instance.isConnected())
+        assert.ok(!AuthUtil.instance.isConnectionExpired())
+    })
+
+    describe('tab and esc', function () {
+        it('single accept', async function () {
+            assertSessionClean()
+            const editor = await openATextEditorWithText('', 'test.py')
+
+            await manualTrigger(editor, client, config)
+            await waitUntilSuggestionSeen()
+            await acceptByTab()
+            // assert.strictEqual(editor.document.getText(), 'Foo')
```

Comment:
comment out these assertions because there is latency between the editor actually updates the document content. and the latency depends on OS and since the purpose of our tests focus on the telemetry, we don't really need these assertions.

Reply:
maybe I will just remove these assertions

Reply:
`assertTextEditorContains` uses waitUntil, doesn't it work? try setting `interval` to a higher number so that waitUntil allows node to do other things. https://github.com/aws/aws-toolkit-vscode-staging/blob/a19a4aaf350ac18ddda0e1a0998afd9f865b0b2c/src/test/testUtil.ts#L264-L271

Reply:
I see, I haven't tried it with a high number, let me try with that

Reply:
I think I will simply remove these assertions of document content, it's still failing on some machines like mac and windows but succeeded on linux